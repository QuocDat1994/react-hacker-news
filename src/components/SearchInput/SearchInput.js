import React from "react";
import "./SearchInput.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";

export default class SearchInput extends React.Component {
  render() {
    return (
      <form className="SearchInput">
        <FontAwesomeIcon icon={faSearch} />
        <input
          name="search"
          type="text"
          value={this.props.value}
          onChange={this.props.onChange}
          placeholder="Search"
        />
        <button type="submit" onClick={this.props.onClick}>
          Search
        </button>
      </form>
    );
  }
}
