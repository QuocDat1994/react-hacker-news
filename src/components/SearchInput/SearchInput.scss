.SearchInput {
  position: relative;
  margin-left: auto;

  svg {
    color: rgba(0, 0, 0, 0.3);
    position: absolute;
    top: 50%;
    left: 10px;
    transform: translateY(-50%);
  }

  input {
    height: 35px;
    font-size: 14px;
    border-radius: 100px;
    border: none;
    outline: none;
    padding-left: 35px;
    padding-right: 15px;
    box-sizing: border-box;
    -webkit-box-shadow: 0px 0px 10px 0px rgba($color: #000000, $alpha: 0.1);
    -moz-box-shadow: 0px 0px 10px 0px rgba($color: #000000, $alpha: 0.1);
    box-shadow: 0px 0px 10px 0px rgba($color: #000000, $alpha: 0.1);
  }

  button {
    display: none;
  }
}
