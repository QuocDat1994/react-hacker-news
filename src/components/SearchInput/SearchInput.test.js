import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SearchInput from "./SearchInput";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchInput", () => {
  it("should render without error", () => {
    const wrapper = shallow(<SearchInput />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without error", () => {
    const wrapper = mount(<SearchInput value="Test" onChange={jest.fn()} />);
    expect(wrapper.find("input").props().value).toBe("Test");
  });
});
