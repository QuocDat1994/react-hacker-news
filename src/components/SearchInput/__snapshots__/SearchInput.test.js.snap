// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render without error 1`] = `ShallowWrapper {}`;

exports[`Loading should render without error 2`] = `ShallowWrapper {}`;

exports[`SearchInput should render without error 1`] = `ShallowWrapper {}`;

exports[`SearchInput should render without error 2`] = `ShallowWrapper {}`;
